### ISSUE TITLE
Deprecate auto_mkdir=True for LocalFileSystem

### ISSUE THREAD

**Issue by TomAugspurger** [intake/filesystem_spec#212]
xref https://github.com/intake/filesystem_spec/pull/211 and https://github.com/dask/dask/issues/5526.

This is a bit of a strange API, that's no longer needed after #211. I'd like to change the default from True to `None`. Users will need to specify `auto_mkdir=True` to retain the old behavior, or `=False` to opt into the new behavior now.
